test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -4,281 +4,234 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
+          AssetManager::AttachmentUpdater.call(attachment.attachment_data, replacement_id: true)
         end
+      end
 
-        context "but replacement has one asset" do
-          let(:sample_rtf) { File.open(fixture_path.join("sample.rtf")) }
-          let(:replacement) { AttachmentData.create!(file: sample_rtf) }
+      context "and the attachment has been replaced with an attachment that has no assets" do
+        it "does not update asset manager" do
+          replacement = AttachmentData.create!(file: File.open(fixture_path.join("whitepaper.pdf")))
+          replacement.assets = []
 
-          it "updates all assets (of attachment to be updated) with original asset ID of replacement attachment" do
-            replacement.assets = [replacement_original_asset]
+          attachment.attachment_data.replace_with!(replacement)
 
-            update_service.expects(:call)
-                          .with(original_asset.asset_manager_id, attachment_data, nil, replacement_attributes)
-            update_service.expects(:call)
-                          .with(thumbnail_asset.asset_manager_id, attachment_data, nil, replacement_attributes)
+          AssetManager::AssetUpdater.expects(:call).never
 
-            updater.call(attachment_data, replacement_id: true)
-          end
+          AssetManager::AttachmentUpdater.call(attachment.attachment_data, replacement_id: true)
         end
+      end
 
-        context "but replacement has no assets" do
-          let(:sample_rtf) { File.open(fixture_path.join("sample.rtf")) }
-          let(:replacement) { AttachmentData.create!(file: sample_rtf) }
+      context "and the attachment has been deleted" do
+        it "does not update the asset" do
+          attachment.delete
 
-          it "does not update asset manager" do
-            update_service.expects(:call).never
+          AssetManager::AssetUpdater.expects(:call).never
 
-            updater.call(attachment_data, replacement_id: true)
-          end
+          AssetManager::AttachmentUpdater.call(attachment.attachment_data, redirect_url: true, draft_status: true)
         end
       end
+    end
 
-      context "when attachment is not synced with asset manager" do
-        let(:sample_rtf) { File.open(fixture_path.join("sample.rtf")) }
-        let(:sample_docx) { File.open(fixture_path.join("sample.docx")) }
-        let(:attachment_data) { AttachmentData.create!(file: sample_rtf, replaced_by: replacement) }
-        let(:replacement) { AttachmentData.create!(file: sample_docx) }
-
-        it "raises a AssetNotFound error" do
-          attachment_data.assets = [thumbnail_asset]
-          attachment_data.assets.create!(asset_manager_id: "asset_manager_id", variant: original, filename: "sample.rtf")
-          replacement.assets.create!(asset_manager_id: "replacement_asset_manager_id", variant: original, filename: "sample.docx")
+    context "when the attachment's attachable is unpublished" do
```

Comment:
Most of this context now wraps the replacement logic tests but I don't think there is a strict correlation there.